locksmithing.lic

```diff
@@ -90,35 +95,49 @@ class Locksmithing
 
   def pick_box(trainer)
     while (DRSkill.getxp('Locksmithing') < 34 && DRC.right_hand)
-      case bput("pick my #{trainer}", 'Maybe you should close', 'not making any progress', "why bother", "it opens.", "isn't locked", 'The lock feels warm', 'The lock looks weak', 'Pick what', 'You need some type of tool to pick', "But you aren't holding")
+      case DRC.bput("pick my #{trainer}", 'Maybe you should close', 'not making any progress', "why bother", "it opens.", "isn't locked", 'The lock feels warm', 'The lock looks weak', 'Pick what', 'You need some type of tool to pick', "But you aren't holding")
       when /You need some type of tool to pick/
-        echo ('YOU HAVE NO LOCKPICKS ON YOUR LOCKPICK RING/BELT.  SORT THAT OUT!')
+        DRC.message('YOU HAVE NO LOCKPICKS ON YOUR LOCKPICK RING/BELT.  SORT THAT OUT!')
         exit
       when /(you should close)|(it opens)|(isn't locked)/
-        bput("close my #{trainer}", 'You close', 'already closed', "You can't close that")
-        bput("lock my #{trainer}", 'You quickly lock', 'already locked')
+        DRC.bput("close my #{trainer}", 'You close', 'already closed', "You can't close that")
+        DRC.bput("lock my #{trainer}", 'You quickly lock', 'already locked')
       when /The lock feels warm/
-        echo('Charges Burned!, checking for other settings.')
+        DRC.message('Charges Burned!, checking for other settings.')
         return
       when /The lock looks weak/
-        case bput("study my #{trainer}", /0 more times/,/risk breaking the lock/,'Study what')
+        case DRC.bput("study my #{trainer}", /0 more times/, /risk breaking the lock/, 'Study what')
         when /0 more times/
-          bput("drop my #{trainer}", 'You drop')
+          DRC.bput("drop my #{trainer}", 'You drop')
           consumable_lockbox
         end
       when /But you aren't holding/
-          echo("Description: '#{trainer}' may not be specific enough or is in a wrong container - something is interfering.")
+          DRC.message("Description: '#{trainer}' may not be specific enough or is in a wrong container - something is interfering.")
           exit
       when /why bother/
-        echo ("The #{trainer} is not a valid option but shares a noun - move it out of your burgle bag to avoid hangups.")
-        bput('stow', 'You')
+        DRC.message("The #{trainer} is not a valid option but shares a noun - move it out of your burgle bag to avoid hangups.")
+        DRC.bput('stow', 'You')
         consumable_lockbox
       when /Pick what/
         return
       end
     end
   end
 
+  def get_item(item, container = nil)
+    return unless item
+    return if DRCI.in_hands?(item)
+    unless DRCI.get_item(item, container)
+      DRC.message("Could not get '#{item}'!")
+      do_exit
+    end
+  end
+
+  def stow_item(item, container = nil)
```

Comment:
Is this (or `get_item`) a helper you'll be adding to many scripts? Perhaps it belongs in `common-something`...

Reply:
Good observation, and I think it's trending that way that they'll be reusable entrypoints for the more sophisticated common-items functions.

The functions went through a few refactorings:
* get_lockpick
* get_lockpick(container)
* ...
* get_item(item, container)

Seems like next step in the evolution is to call `DRCI.get_item(item, container)` and `DRCI.stow_item(item, container)`